internal/eval/deque.go

```diff
@@ -215,6 +302,179 @@ func (q *Deque) RPop() (string, error) {
 	return x, nil
 }
 
+func (q *Deque) breakPivotNodeAndInsertAfter(qIterator *DequeIterator, pivotNode *byteListNode, element string) *byteListNode {
+	newNode := q.list.newNode()
+	if pivotNode.next != nil {
+		pivotNode.next.prev = newNode
+	}
+	newNode.next = pivotNode.next
+	pivotNode.next = nil
+	newNode.buf = append([]byte{}, pivotNode.buf[qIterator.BufIndex:]...)
+	pivotNode.buf = pivotNode.buf[:qIterator.BufIndex]
+	q.list.tail = pivotNode
+	q.RPush(element)
+	newNode.prev = q.list.tail
+	q.list.tail.next = newNode
+	return newNode
+}
+
+func (q *Deque) insertAfterPivotNodeHelper(element string, qIterator *DequeIterator, pivotNode *byteListNode) {
```

Comment:
Add a comment explaining what this method does.

Reply:
Done.